data/keystore/src/test/java/tech/pegasys/artemis/bls/keystore/KeyStoreTest.java

```diff
@@ -0,0 +1,262 @@
+    final Bytes largeData = Bytes.random(1024);
+    final Crypto crypto =
+        KeyStore.encryptUsingCipherFunction(
+            largeData, PASSWORD, kdfParam, cipher, ChecksumFunction.SHA256);
+    assertEquals(32, crypto.getChecksum().getMessage().size());
+  }
+
+  @ParameterizedTest
+  @EnumSource(CryptoFunction.class)
+  void encryptSmallDataWithRandomSaltAndIv(final CryptoFunction cryptoFunction) {
+    final KdfParam kdfParam =
+        cryptoFunction == SCRYPT
+            ? SCryptParamBuilder.aSCryptParam().build()
+            : Pbkdf2ParamBuilder.aPbkdf2Param().build();
+    final CipherParam cipherParam = CipherParamBuilder.aCipherParam().build();
+    final Cipher cipher =
+        CipherBuilder.aCipher()
+            .withCipherParam(cipherParam)
+            .withCipherFunction(AES_128_CTR)
+            .build();
+    final Bytes smallData = Bytes.random(8);
+    final Crypto crypto =
+        KeyStore.encryptUsingCipherFunction(
+            smallData, PASSWORD, kdfParam, cipher, ChecksumFunction.SHA256);
+    assertEquals(32, crypto.getChecksum().getMessage().size());
+  }
+
+  @ParameterizedTest
+  @EnumSource(CryptoFunction.class)
+  void checksumFunctionSha512(final CryptoFunction cryptoFunction) {
+    final KdfParam kdfParam =
+        cryptoFunction == SCRYPT
+            ? SCryptParamBuilder.aSCryptParam().build()
+            : Pbkdf2ParamBuilder.aPbkdf2Param().build();
+    final CipherParam cipherParam = CipherParamBuilder.aCipherParam().build();
+    final Cipher cipher =
+        CipherBuilder.aCipher()
+            .withCipherParam(cipherParam)
+            .withCipherFunction(AES_128_CTR)
+            .build();
+    final Bytes smallData = Bytes.random(8);
+    final Crypto crypto =
+        KeyStore.encryptUsingCipherFunction(
+            smallData, PASSWORD, kdfParam, cipher, ChecksumFunction.SHA512);
+    assertEquals(64, crypto.getChecksum().getMessage().size());
+  }
+
+  @ParameterizedTest
+  @EnumSource(CryptoFunction.class)
+  void encryptionWithInvalidDkLenShouldFail(final CryptoFunction cryptoFunction) {
+    final KdfParam kdfParam =
+        cryptoFunction == SCRYPT
+            ? SCryptParamBuilder.aSCryptParam().withDklen(15).build()
+            : Pbkdf2ParamBuilder.aPbkdf2Param().withDklen(15).build();
+    final CipherParam cipherParam = CipherParamBuilder.aCipherParam().build();
+    final Cipher cipher =
+        CipherBuilder.aCipher()
+            .withCipherParam(cipherParam)
+            .withCipherFunction(AES_128_CTR)
+            .build();
+    final Bytes data = Bytes.random(8);
+    final IllegalArgumentException illegalArgumentException =
+        assertThrows(
+            IllegalArgumentException.class,
+            () ->
+                KeyStore.encryptUsingCipherFunction(
+                    data, PASSWORD, kdfParam, cipher, ChecksumFunction.SHA256));
+    assertEquals(
+        "aes-128-ctr requires kdf dklen greater than 16", illegalArgumentException.getMessage());
```

Comment:
We should be using assertj for all these assertions.  At which point you can use:
`Assertions.assertThatThrownBy(() -> ...).isInstanceOf(IllegalArgumentException.class).hasMessage("...")`

All the `assertEquals` should be `assertThat(...).isEqualTo(...)` as well.

Reply:
Introduced assert4j.